Merge Branch-UG HL with master (#33)
AY2223S1-CS2103-F09-2/tp: docs/UserGuide.md

---
layout: page
title: User Guide
---

> AddressBook Level 3 (AB3) is a **desktop app for managing contacts, optimized for use via a Command Line Interface** (CLI) while still having the benefits of a Graphical User Interface (GUI). If you can type fast, AB3 can get your contact management tasks done faster than traditional GUI apps.

Reviewer: The existing user guide seems to have been deleted in your copy, could you please try adding your changes within the existing user guide?

> ### Format: ```add n/NAME p/PHONE_NUMBER e/EMAIL a/ADDRESS [d/DOB]```

Reviewer: before ###Format, we'd need a line __Optional Parameter 2: Date of Birth__ for consistency with Qiao Ran

> - Adds date of birth to a person in the contact list. Date formats accepted are: d.M.yyyy , dd MMM yyyy, M d, yyyy

Reviewer: TBH it took me a while to understand the date formats, do you think dd.mm.yyyy would better represent 20.03.2000?
Also, month dd, yyyy to represent March 20, 2000 & keeping the current dd MMM yyyy for dd MMM yyyy? What do ya think?

Author: mmmm sounds gd! will change

### Examples:
- ```add n/John Wang p/98765432 e/johnwang@example.com a/John street, block 123, #01-01 d/20.03.2000```
- ```add n/John p/92781123 e/john@example.com a/Donald street, block 248, #02-04 d/20 MAR 2000```
- ```add n/Charlotte p/81286623 e/charlotte@example.com a/Charity street, block 101, #10-82 d/March 20, 2000```

> ## Optional Parameter 2: Date of Birth

Reviewer: For this part, can discuss with Qiao Ran, on standardising with underscores __ or hashtag symbol, although personally I think underscores may be help differentiate it better & since its a subset of a triple hashtag ### pointer, I suppose 4 hastags would work better

### Format: ```edit INDEX [n/NAME] [d/DOB]```

- Edits the date of birth of a person in the contact list. Date formats accepted are: d.M.yyyy, dd MMM yyyy, M d, yyyy
- INDEX must be positive integers (i.e 1,2,3…)
- INDEX must be within the range of the contact list index (i.e from 1 to size of contact list).

### Examples:
- ```edit 1 d/20.03.2000```
- ```edit 2 d/20 MAR 2000```

> - ```edit 3 d/March 20, 2000```

Reviewer: A newline would be necessary here to prevent codecov from failing!

Author: i see thank you for letting me know

* `add n/John p/92781123 e/john@example.com a/Donald street, block 248, #02-04 g/m`
* `add n/Charlotte p/81286623 e/charlotte@example.com a/Charity street, block 101, #10-82  g/F`

__Optional Parameter 2: Date of Birth__

Format: `add n/NAME p/PHONE_NUMBER e/EMAIL a/ADDRESS [d/DOB]`

> `

Reviewer: Additional ` can be removed
```suggestion

```

Author: ah thx for catching that

> - Adds date of birth to a person in the contact list. Date formats accepted are: 		dd.MM.yyyy , dd MMM yyyy, M d, yyyy

Reviewer: Perhaps, can remove the huge whitespace between are: and dd.MM.yyy & use ; to separate the formats since comma is already used in the third format
```suggestion
- Adds date of birth to a person in the contact list. Date formats accepted are: dd.MM.yyyy ; dd MMM yyyy ; M d, yyyy
```

Author: thanks for catching this too

* `edit 2 g/f`
* `edit 3 g/F`

__Optional Parameter 2: Date of Birth__

Format: `edit INDEX [d/DOB]`

> - Edits the date of birth of a person in the contact list. Date formats accepted are: d.M.yyyy, dd MMM yyyy, M d, yyyy

Reviewer: Ah it needs the ; separator here as well for the 3 types of formats

Author: ok added! changed the date formats as previously stated as well sry forgot to add that in

* `edit 2 g/f`
* `edit 3 g/F`

__Optional Parameter 2: Date of Birth__

> Format: `edit INDEX [n/NAME] [d/DOB]`

Reviewer: I think the [d/NAME] may not be necessary here? Especially since we are focusing on editing the DOB only!
```suggestion
Format: `edit INDEX [d/DOB]`
```

- Edits the date of birth of a person in the contact list. Date formats accepted are: d.M.yyyy, dd MMM yyyy, M d, yyyy

> - `INDEX must be positive integers (i.e 1,2,3…)`
> - `INDEX must be within the range of the contact list index (i.e from 1 to size of contact list).`

Reviewer: Would be better to format only the INDEX and not the whole line! Also, to standardise with QR & since its singular, might want to use the term 'a' positive integer
```suggestion
- `INDEX` must be **a positive integer** (i.e 1,2,3…)
- `INDEX` must be within the range of the contact list index (i.e from 1 to size of contact list).
```

Author: mmmm ic!
